File: backend/src/main/java/com/yassine/backend/Dao/Offre.java
package com.yassine.backend.Dao;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.sql.Date;

@Entity @Data @AllArgsConstructor @NoArgsConstructor @ToString
public class Offre {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idOffre;

    private Date dateDebut;
    private String libelle;
    private String description;
    private Date dateFin;

    @ManyToOne
    private Utilisateur utilisateur;

}

File: frontend/lib/models/Offre.dart
class Offre {
  final int idOffre;
  final String dateDebut;
  final String libelle;
  final String description;
  final String dateFin;

  Offre({
    required this.idOffre,
    required this.dateDebut,
    required this.libelle,
    required this.description,
    required this.dateFin,
  });

  factory Offre.fromJson(Map<String, dynamic> json) {
    return Offre(
      idOffre: json['idOffre'],
      dateDebut: json['dateDebut'],
      libelle: json['libelle'],
      description: json['description'],
      dateFin: json['dateFin'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idOffre': idOffre,
      'dateDebut': dateDebut,
      'libelle': libelle,
      'description': description,
      'dateFin': dateFin,
    };
  }
}
